src/Tasks/ResourceHandling/MSBuildResXReader.cs

```diff
@@ -234,10 +233,46 @@ private static void AddLinkedResource(string resxFilename, bool pathsRelativeToB
                     return;
                 }
             }
+            else if (IsByteArray(fileRefType))
+            {
+                byte[] byteArray = File.ReadAllBytes(fileName);
+
+                resources.Add(new LiveObjectResource(name, byteArray));
+                return;
+            }
+            else if (IsMemoryStream(fileRefType))
+            {
+                // See special-case handling in ResXFileRef
+                // https://github.com/dotnet/winforms/blob/689cd9c69e632997bc85bf421af221d79b12ddd4/src/System.Windows.Forms/src/System/Resources/ResXFileRef.cs#L293-L297
+                byte[] byteArray = File.ReadAllBytes(fileName);
+
+                resources.Add(new LiveObjectResource(name, new MemoryStream(byteArray)));
+                return;
+            }
 
             resources.Add(new FileStreamResource(name, fileRefType, fileName, resxFilename));
         }
 
+        /// <summary>
+        /// Does this assembly-qualified type name represent an array of bytes?
+        /// </summary>
+        /// <remarks>
+        /// We can't hard-code byte[] type name due to version number
+        /// updates & potential whitespace issues with ResX files.
+        ///
+        /// Comment and logic from https://github.com/dotnet/winforms/blob/16b192389b377c647ab3d280130781ab1a9d3385/src/System.Windows.Forms/src/System/Resources/ResXDataNode.cs#L411-L416
+        /// </remarks>
+
+        private static bool IsByteArray(string fileRefType)
+        {
+            return fileRefType.IndexOf("System.Byte[]") != -1 && fileRefType.IndexOf("mscorlib") != -1;
```

Comment:
NIT: Should you have variables for these strings, like you have on IsMemoryStream?

Reply:
For this, I'm preserving the code as it was in the original ResXReader code in the comment above. Would you prefer static strings? I'm not opposed.

Reply:
Not super worried about it.

Reply:
(Realize this is preserved, but sure is hacky that this is just a couple of indexofs that would have all sorts of false matches in theory.)

Reply:
Yeah, its primary purpose these days is to make everyone who looks at it sad :(